### PR TITLE
RELATED: BB-1073 Support execution of AFM with multiple date datasets…

### PR DESCRIPTION
**Pull request by tomassatinsky**
… in global and measure date filters

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-js-opensource-repository-check/49/ : SUCCESS in 10s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-js-unit-tests-zuul-docker-el7/45/ : SUCCESS in 1m 20s

File: src/DataLayer/helpers/filters.ts [hunk 1]
```diff
@@ -55,15 +59,18 @@ function getGlobalDateFilters(afm: AFM.IAfm): AFM.DateFilterItem[] {
...
- * 1. Remove date filter (D1) from global
- * 2. Add D1 to each metric without date filter
- * 3. M1 is untouched
+ * When:
+ *   There is at least one metric with date filter & global date filter present
+ * Then:
+ *   1. To each metric add all global date filters if there isn't other date filter with the same date data set
+ *   2. Remove global date filter
+ * Otherwise
+ *   Return provided AFM without any chang
```
Comment (martinmosko) [inline src/DataLayer/helpers/filters.ts]: typo `chang`

File: src/DataLayer/helpers/tests/filters.spec.ts [hunk 2]
```diff
@@ -159,152 +159,158 @@ describe('mergeFilters', () => {
...
-                            dataSet: {
-                                identifier: 'd2'
-                            },
-                            from: '1',
-                            to: '2'
-                        }
-                    }
-                ]
-            }
+    const dateFilterD1A: AFM.DateFilterItem = {
```
Comment (martinmosko) [inline src/DataLayer/helpers/tests/filters.spec.ts]: D1A its not quite straightforward what it is, please could you pick some more explicit name ? :)

File: src/DataLayer/helpers/tests/filters.spec.ts [hunk 3]
```diff
@@ -159,152 +159,158 @@ describe('mergeFilters', () => {
...
+            granularity: 'day',
+            from: 3,
+            to: 4
+        }
+    };
+
+    const dateFilterD2A: AFM.DateFilterItem = {
+        absoluteDateFilter: {
+            dataSet: {
+                identifier: 'd2'
```
Comment (martinmosko) [inline src/DataLayer/helpers/tests/filters.spec.ts]: try 'uri' as well :)

File: src/DataLayer/utils/AfmUtils.ts [hunk 4]
```diff
@@ -210,7 +210,14 @@ export function getId(obj: AFM.ObjQualifier): string | null {
...
 
-function getDateFilterDateDataSet(filter: AFM.DateFilterItem): AFM.ObjQualifier | null {
+/**
+ * Returns date filter date dataset
+ *
+ * @method getDateFilterDateDataSet
+ * @param {AFM.DateFilterItem} filter
+ * @returns {AFM.ObjQualifier | null }
+ */
+export function getDateFilterDateDataSet(filter: AFM.DateFilterItem): AFM.ObjQualifier | null {
```
Comment (martinmosko) [inline src/DataLayer/utils/AfmUtils.ts]: 'null'

Comment (martinmosko) [reply src/DataLayer/utils/AfmUtils.ts]: Write a test to cover case with thrown Error

File: src/DataLayer/helpers/tests/filters.spec.ts [hunk 5]
```diff
@@ -159,152 +159,158 @@ describe('mergeFilters', () => {
...
+                    item: {
+                        identifier
+                    },
+                    filters
+                }
+            }
+        };
+    }
+
+    it('should move global date filters to measure date filters if dataset of the filter is not equal', () => {
```
Comment (martinmosko) [inline src/DataLayer/helpers/tests/filters.spec.ts]: dataSet

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-js-opensource-repository-check/52/ : SUCCESS in 13s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-js-unit-tests-zuul-docker-el7/49/ : SUCCESS in 1m 40s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-js-opensource-repository-check/53/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-js-unit-tests-zuul-docker-el7/50/ : SUCCESS in 43s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-js-opensource-repository-check/54/ : SUCCESS in 11s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-js-unit-tests-zuul-docker-el7/51/ : SUCCESS in 50s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-js-merge-compliance-check/13/ : SUCCESS in 15s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-js-unit-tests-zuul-docker-el7/52/ : SUCCESS in 44s

**Pull request merged**